tools/trace_analyzer_tool.cc

```diff
@@ -0,0 +1,1719 @@
+//  (found in the LICENSE.Apache file in the root directory).
+//
+
+#ifndef ROCKSDB_LITE
+
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
+#ifdef GFLAGS
+#ifdef NUMA
+#include <numa.h>
+#include <numaif.h>
+#endif
+#ifndef OS_WIN
+#include <unistd.h>
+#endif
+
+#include <fcntl.h>
+#include <inttypes.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <time.h>
+#include <condition_variable>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+
+#include "db/db_impl.h"
+#include "db/memtable.h"
+#include "db/write_batch_internal.h"
+#include "options/cf_options.h"
+#include "rocksdb/db.h"
+#include "rocksdb/env.h"
+#include "rocksdb/iterator.h"
+#include "rocksdb/slice.h"
+#include "rocksdb/slice_transform.h"
+#include "rocksdb/status.h"
+#include "rocksdb/table_properties.h"
+#include "rocksdb/utilities/ldb_cmd.h"
+#include "rocksdb/write_batch.h"
+#include "table/meta_blocks.h"
+#include "table/plain_table_factory.h"
+#include "table/table_reader.h"
+#include "tools/trace_analyzer_tool_imp.h"
+#include "util/coding.h"
+#include "util/compression.h"
+#include "util/file_reader_writer.h"
+#include "util/gflags_compat.h"
+#include "util/random.h"
+#include "util/string_util.h"
+#include "util/trace_replay.h"
+
+using GFLAGS_NAMESPACE::ParseCommandLineFlags;
+using GFLAGS_NAMESPACE::RegisterFlagValidator;
+using GFLAGS_NAMESPACE::SetUsageMessage;
+
+DEFINE_string(trace_path, "", "The trace file path.");
+DEFINE_string(output_dir, "", "The directory to store the output files.");
+DEFINE_string(output_prefix, "", "The prefix used for all the output files.");
+DEFINE_bool(output_key_stats, false,
+            "Output the key access count statistics to file\n"
+            "for accessed keys:\n"
+            "format:[cf_id value_size access_keyid access_count]\n"
+            "for the whole key space keys:\n"
+            "format:[whole_key_space_keyid access_count]");
+DEFINE_bool(output_access_count_stats, false,
+            "Output the access count distribution statistics to file."
+            "format:[access_count number_of_access_count]");
+DEFINE_uint64(output_time_series, 0,
+              "trace collect time, in microseconds"
+              "Output the access time sequence of each key\n"
+              "format:[type_id time_in_sec access_key_id].");
+DEFINE_int32(output_prefix_cut, 0,
+             "The number of bytes as prefix to cut the keys."
+             "if it is enabled, it will generate the following:\n"
+             "for accessed keys:\n"
+             "format:[acessed_keyid access_count num_keys ave_access prefix]\n"
+             "for whole key space keys:\n"
+             "format:[start_keyid_in_whole_keyspace prefix]\n"
+             "if 'output_qps_stats' is enabled, it will output:\n"
+             "format:[time_in_sec IO_num], [prefix qps_of_this_second].");
+DEFINE_bool(output_trace_sequence, false,
```

Comment:
1. May be rename it to something like `convert_to_human_readable_trace`? Otherwise I think people might have a hard time finding this option. 
2. Could you also specify the filename where the converted output goes to?